### ISSUE TITLE
No Controls on Swim Spa

### ISSUE THREAD

**Issue by mike240se** [garbled1/balboa_homeassistan#26]
First of all, thanks so much for this.   Was really excited when I saw this in the announcement for HA 2021.12.   I have a swim spa with 2 swim jet pumps, the seat pump, and a recirc pump but none of the pumps are available as status or able to control however they are controlable in the BWA app.

https://i.imgur.com/DERpR0l.png

Thanks

**Comment by garbled1** [comment]
Yes, currently the HA version doesn't have a few of the platforms integrated yet.  In HA, you have to do one platform at a time, so I did the binary_sensor and the climate ones first.  My plan is to keep tackling them one at a time, until the whole set is in, and then deprecate this HACS module.

Basically, if you need full operation immediately, you can switch to the HACS module, until everything is ported over.  Or if you are just patient, everything should show up in the next 1-2 releases of HA, and you should be good.  Glad to see what I have is working for you though..  Hopefully it won't be long now.

**Comment by mike240se** [comment]
> Yes, currently the HA version doesn't have a few of the platforms integrated yet. In HA, you have to do one platform at a time, so I did the binary_sensor and the climate ones first. My plan is to keep tackling them one at a time, until the whole set is in, and then deprecate this HACS module.
> 
> Basically, if you need full operation immediately, you can switch to the HACS module, until everything is ported over. Or if you are just patient, everything should show up in the next 1-2 releases of HA, and you should be good. Glad to see what I have is working for you though.. Hopefully it won't be long now.

Awesome! I can probably wait but I will also check out the HACS version in the meantime!   Thanks much.

**Comment by lowema** [comment]
Just found this ... thanks for the clarification :)

**Comment by gazoodle** [comment]
Totally off-topic, I know, but I’ve built essentially the same stuff as you @garbled1 for Gecko powered hot tubs … a python protocol lib and a HACS integration, and I’m wondering what your thoughts/experiences were on taking your integration to a native HA one? I’ve reviewed much of your code and saw you had got some platforms over but you’ve not had much activity since then. Is this down to how busy you are, or the HA team, or perhaps you’re disillusioned now … ? 😀

Quick links
[https://github.com/gazoodle/geckolib](url)
[https://github.com/gazoodle/gecko-home-assistant](url)
